ethcore/src/engines/authority_round/mod.rs

```diff
@@ -800,11 +982,17 @@ impl Engine<EthereumMachine> for AuthorityRound {
 				chain(hash).and_then(|header| {
 					if header.number() == 0 { return None }
 
-					let res = (hash, header.author().clone());
-					trace!(target: "finality", "Ancestry iteration: yielding {:?}", res);
+					if let Ok(signers) = header_signers(&header) {
+						let res = (hash, signers);
```

Comment:
this will import into `RollingFinality` as though the empty step signers have signed on the hash of the current block, but it's actually wrong: the empty step signers have only signaled their approval of the parent header.

Reply:
it will probably be easier to alter the iterator logic than to alter the rolling finality checker

Reply:
Yes that's true, the empty step signers only signed on the parent block. The reason I altered the rolling finality checker was to make sure it didn't return duplicate hashes.

Reply:
I think the current alterations are good actually, just that we need to figure out the correct way to input. I think that's one of the pieces of code that's most important to have tests for, especially if we have entry merging logic